src/Field/Choices/StoryChoices.php

```diff
@@ -2,22 +2,22 @@
 
 namespace Torr\Storyblok\Field\Choices;
 
-use Torr\Storyblok\Component\Reference\ComponentsWithTags;
+use Torr\Storyblok\Component\Filter\ComponentFilter;
 use Torr\Storyblok\Context\ComponentContext;
 use Torr\Storyblok\Field\Data\Story\StoryReferenceList;
 use Torr\Storyblok\Field\Data\StoryReferenceData;
+use Torr\Storyblok\Manager\Sync\Filter\ResolvableComponentFilter;
 
 /**
  * Makes a story selectable
  */
 final class StoryChoices implements ChoicesInterface
 {
 	/**
-	 * @param array<string>|ComponentsWithTags $restrictContentTypes
-	 * @param string|\BackedEnum|null          $referencedStoryDataMode This is a Key that will be used by the corresponding StoryNormalizer to find out which data from the referenced Story is needed by the component
+	 * @param string|\BackedEnum|null $referencedStoryDataMode This is a Key that will be used by the corresponding StoryNormalizer to find out which data from the referenced Story is needed by the component
 	 */
 	public function __construct (
-		private readonly array|ComponentsWithTags $restrictContentTypes = [],
+		private readonly ComponentFilter $components = new ComponentFilter(),
```

Comment:
Nit: Should we rename this field to `$filteredComponents`? I feel like `$components` is a bit misleading.

Reply:
I was going back and forth on this one, not sure whether we should keep it consistent or use different names.

Naming suggestions:
- `restrictToComponents`
- `allowedComponents`

Reply:
Are renamed, please take a look @keichinger